### ISSUE TITLE
System reverts formatting in Methods section

### ISSUE THREAD

**Issue by bryanmgee** [CDL-Dryad/dryad-product-roadmap#2572]
**Issue**: Two curators have independently encountered datasets where removing the formatting of the text within the Methods field is not saved by the system, which repeatedly reverts to the original formatting as input by the author, regardless of whether the 'Tab' key is hit after making the change or by manually clicking into another field. This also includes if the text is copied into Microsoft Word, reformatted, and copied back in.

**Examples:** 

- [Dataset 1](https://datadryad.org/stash/ds_admin?q=%2210.5061%2Fdryad.w3r2280wn%22&sort=relevance&commit=Search&tenant=&editor_id=&curation_status=&publication_name=&sponsor_org=)
- [Dataset 2](https://datadryad.org/stash/ds_admin?q=%2210.5061%2Fdryad.zgmsbccb6%22&sort=relevance&commit=Search&tenant=&editor_id=&curation_status=&publication_name=&sponsor_org=&utf8=%E2%9C%93)
- [Dataset 3](https://datadryad.org/stash/ds_admin?q=%2210.5061%2Fdryad.jsxksn0g0%22&sort=relevance&commit=Search&tenant=&editor_id=&curation_status=&publication_name=&sponsor_org=&utf8=%E2%9C%93) [added 6/7]
- [Dataset 4](https://datadryad.org/stash/ds_admin?q=%2210.5061%2Fdryad.q573n5tph%22&sort=relevance&commit=Search&tenant=&editor_id=&curation_status=&publication_name=&sponsor_org=&utf8=%E2%9C%93) [added 6/7, this one is only about the extra line break at the end]
- [Dataset 5](https://datadryad.org/stash/ds_admin?q=%2210.5061%2Fdryad.47d7wm3gx%22&sort=relevance&commit=Search&tenant=&editor_id=&curation_status=&publication_name=&sponsor_org=&utf8=%E2%9C%93) [added 6/8]
- [Dataset 6](https://datadryad.org/stash/dataset/doi:10.5061/dryad.5hqbzkhbk) [added 6/8]
- [Dataset 7](https://datadryad.org/stash/ds_admin?q=%22BLM+overexpression+as+a+predictive+biomarker+for+CHK1+inhibitor+response+in+PARP+inhibitor%E2%80%93resistant+BRCA-mutant+ovarian+cancer%22&sort=relevance&commit=Search&tenant=&editor_id=&curation_status=&publication_name=&sponsor_org=&utf8=%E2%9C%93) [added 6/9]
- [Dataset 8](https://datadryad.org/stash/ds_admin?q=%22TLR7%2F8+stress+response+drives+histiocytosis+in+SLC29A3+disorders%22&sort=relevance&commit=Search&curation_status=&editor_id=&publication_name=&sponsor_org=&tenant=&utf8=%E2%9C%93) [added 6/13]
- [Dataset 9](https://datadryad.org/stash/ds_admin?q=%2210.5061%2Fdryad.v9s4mw71n+%22&sort=relevance&commit=Search&tenant=&editor_id=&curation_status=&publication_name=&sponsor_org=&utf8=%E2%9C%93) [added 6/15]
- [Dataset 10](https://datadryad.org/stash/ds_admin?q=%22Analysis+of+the+Proteomic+Profile+in+Serum+of+Irradiated+Nonhuman+Primates+Treated+with+Ex-Rad%2C+a+Radiation+Medical+Countermeasure%22&sort=relevance&commit=Search&tenant=&editor_id=&curation_status=&publication_name=&sponsor_org=&utf8=%E2%9C%93) [added 6/27]
- [Dataset 11](https://datadryad.org/stash/ds_admin?q=%2210.5061%2Fdryad.vdncjsxzn%22&sort=relevance&commit=Search&tenant=&editor_id=&curation_status=&publication_name=&sponsor_org=&utf8=%E2%9C%93) [added 6/27]

Based on what I have tested out so far, if you delete the entire contents of the Methods field and enter whatever sort of text, it will go back to using the default formatting. This does not occur with all datasets, and I haven't been able to discern a pattern in which datasets this happens to.

**Comment by sfisher** [comment]
I am not able to duplicate this problem.  Do you have any idea of steps to replicate?  I can change only the formatting and it save the changes.

![Image](https://github.com/CDL-Dryad/dryad-product-roadmap/assets/105433/16861db2-192a-4532-88d3-2c54b2f80a3c)


![Image](https://github.com/CDL-Dryad/dryad-product-roadmap/assets/105433/863de23e-e913-40b9-96f6-b2b8889277cc)

Is it specific to any particular browser or with any commonalities in the content that you can see? I'm afraid this will be hard to fix if it cannot be duplicated.  The methods also uses the same rich text entry as the other two so I wouldn't expect the behavior to be different.  I can look through the code a little for things that might cause problems, but I think it's pretty similar to abstract and usage notes.  Have you ever seen it happen anywhere but the methods?

Is this still happening?

**Comment by bryanmgee** [comment]
Yeah, this is what I was worried about; I have not seen it crop up in about a week or so, although I have gone back to some of the ones that are listed above to see if I can get the reformatting to stick, and it still reverts once you switch to a different page in the submission process. Have not seen it occur for any of the other fields. I at least don't think it's related to the browser that the curator is using since it does not occur consistently; I've tried it on both Chrome and Firefox without plug-ins.

**Comment by sfisher** [comment]
I'll try one of these datasets then though I hate to edit user data.  I guess as long as I can revert it.

**Comment by sfisher** [comment]
Yeah, I see it happening on some of those datasets.  I see it trying to save but getting "forbidden" status.  Weird when I copy the same data to development it seems like it works.  IDK what is happening.  I'll see what I can figure out.

**Comment by bryanmgee** [comment]
Scott, just letting you know that I found another previously uncurated dataset with this bug (added it to the original ticket), so it does seem like this may in fact still be plaguing some datasets.

**Comment by sfisher** [comment]
Thanks, @bryanmgee .  I found the cause of this problem.  Some requests were being blocked for malicious activity, especially when people have copied and pasted complex formatting (especially if it came from Microsoft Word which adds in even more extra crap into the HTML).  Having all the example datasets was really helpful since it was dependent on the text and HTML that is generated and that got flagged by the Amazon Web Application Firewall as malicious PHP code.  (It wasn't, but it maybe sort of looked like it to Amazon's rules?)  When it has hated content it would refuse the requests and not update.

I was able to disable this rule and tested it out on some descriptions that seemed to trigger the WAF and it is no longer blocking these requests.

Let me know if you see more of these or other weird behavior where a page isn't loading or a request isn't going through.  We went through the WAF rules to try to disable things that cause problems, but clearly some things aren't triggered except in unusual circumstances or some unusual cases that our analysis didn't catch.
